composite_controller.go

```diff
@@ -35,14 +34,14 @@ import (
 	utilerrors "k8s.io/apimachinery/pkg/util/errors"
 )
 
-func syncAllCompositeControllers(dynClient *dynamicClientset, metaClient *internalclient.Clientset) error {
-	ccList, err := metaClient.MetacontrollerV1alpha1().CompositeControllers("").List(metav1.ListOptions{})
+func syncAllCompositeControllers(dynClient *dynamicClientset, mc *metaController) error {
+	ccList, err := (mc.ccLister).List(labels.NewSelector())
 	if err != nil {
 		return fmt.Errorf("can't list CompositeControllers: %v", err)
 	}
 
-	for i := range ccList.Items {
-		cc := &ccList.Items[i]
+	for i := range ccList {
```

Comment:
Now that we're already getting a slice of pointers, we can simplify this to:

```go
for _, cc := range ccList {
```

Same for `icList` below.